Enable/View Controllers/HomeViewController.m

```diff
@@ -213,4 +175,47 @@ - (void)setGMSCameraCoordinatesWithLatitude:(CLLocationDegrees)latitude longitud
     [self updateLocationMarkersWithProjection:self.mapView.mapView.projection radius:self.radiusMiles];
 }
 
+#pragma mark - Setup
+- (void) setUpSearch {
+    self.resultsViewController = [[GMSAutocompleteResultsViewController alloc] init];
+    self.searchController = [[UISearchController alloc]
+                                initWithSearchResultsController:self.resultsViewController
+                            ];
+    self.resultsViewController.delegate = self;
+    self.searchController.searchResultsUpdater = self.resultsViewController;
+    [self.searchController setHidesNavigationBarDuringPresentation:NO];
+    
+    UIView *subView = [[UIView alloc] initWithFrame:CGRectZero];
+    subView.translatesAutoresizingMaskIntoConstraints = NO;
+    [self.view addSubview:subView];
+    [subView.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor constant:0].active = YES;
+    [subView.trailingAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.trailingAnchor constant:0].active = YES;
+    [subView.leadingAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.leadingAnchor constant:0].active = YES;
+    [subView.heightAnchor constraintEqualToConstant:50].active = YES;
+    [subView.bottomAnchor constraintEqualToAnchor:self.mapView.topAnchor constant:0].active = YES;
+    
+    [subView addSubview:self.searchController.searchBar];
+    [self.searchController.searchBar sizeToFit];
+}
+
+- (void) setupTheme {
+    [self setupMainTheme];
+    NSLog(@"%@", [ThemeTracker sharedTheme].theme);
+    NSDictionary * colorSet = [ThemeTracker sharedTheme].colorSet;
+    // search bar
+    [self.searchController.searchBar.searchTextField setBackgroundColor:[UIColor colorNamed: colorSet[@"Secondary"]]];
+    [self.searchController.searchBar setBarTintColor:[UIColor colorNamed: colorSet[@"Background"]]];
+    [self.searchController.searchBar setTintColor:[UIColor colorNamed: colorSet[@"Accent"]]];
+    self.searchController.searchBar.searchTextField.attributedPlaceholder = [[NSAttributedString alloc] initWithString:@"Search location..." attributes:@{NSForegroundColorAttributeName: [UIColor colorNamed: colorSet[@"Label"]]}];
+    [self.searchController.searchBar.searchTextField.leftView setTintColor: [UIColor colorNamed: colorSet[@"Accent"]]];
+    [self.searchController.searchBar.searchTextField.rightView setTintColor: [UIColor colorNamed: colorSet[@"Accent"]]];
+    
+    // results view
```

Comment:
Those `// search bar` and `// result view` comments are nice, but you could avoid them by breaking down those two sections in `setupThemeForSeachBar` and `setupThemeForResultView` functions 😉. 

When you feel like you need to add comments to break down your code, it's usually a sign that you should simply break down the function in multiple smaller function!